src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/Win32.cs

```diff
@@ -97,31 +97,27 @@ internal static byte[] ConvertIntPtrSidToByteArraySid(IntPtr binaryForm)
         //
 
 
-        internal static int CreateSidFromString(
+        internal static unsafe int CreateSidFromString(
             string stringSid,
             out byte[]? resultSid
             )
         {
             int ErrorCode;
-            IntPtr ByteArray = IntPtr.Zero;
+            void* pSid = null;
 
             try
             {
-                if (Interop.BOOL.FALSE == Interop.Advapi32.ConvertStringSidToSid(stringSid, out ByteArray))
+                if (Interop.BOOL.FALSE == Interop.Advapi32.ConvertStringSidToSid(stringSid, out pSid))
                 {
                     ErrorCode = Marshal.GetLastPInvokeError();
                     goto Error;
                 }
 
-                resultSid = ConvertIntPtrSidToByteArraySid(ByteArray);
+                resultSid = ConvertIntPtrSidToByteArraySid((IntPtr)pSid);
             }
             finally
             {
-                //
-                // Now is a good time to get rid of the returned pointer
-                //
-
-                Marshal.FreeHGlobal(ByteArray);
+                Interop.Kernel32.LocalFree(pSid);
```

Comment:
>If the hMem parameter is NULL, LocalFree ignores the parameter and returns NULL.

@AaronRobinsonMSFT I've unintentionally removed a null check here by calling `LocalFree` directly, so that's an unnecessary pinvoke on the failure path. Add a null check or not?

Reply:
There is also the Win32 check for Atom values. I don't think the `ConvertStringSidToSidW` API returns an Atom value, but could be mistaken. Is there a compelling reason to not keep using `Marshal.FreeHGlobal`?

Reply:
There's an argument it simplifies the code, and returned a 16-bit legacy type seems unlikely, but there's no need in this PR.